### DISCUSSION TITLE
Руководство ПЛИ, п2.2-2.11 заголовок документа

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#268] [Лабораторная РГ]
### 2.2. Область применения realmCode
У1-1: Один [1..1] элемент ClinicalDocument/realmCode ОБЯЗАН быть представлен со значением атрибута @code равным "RU".
<img width="491" alt="image" src="https://github.com/user-attachments/assets/b88e8325-eb36-455d-b740-232bd42468a4">

Обсуждение [ниже](https://github.com/fhir-ru/core/discussions/268#discussioncomment-10600790)

**Comment by eugenekogan-spb** [comment]
### 2.3. Указатель на тип структуры документа typeId
У1-2: Один [1..1] элемент ClinicalDocument/typeId ОБЯЗАН быть представлен (атрибут @root должен иметь значение "2.16.840.1.113883.1.3", атрибут @extension должен иметь значение "POCD_MT000040").
<img width="766" alt="image" src="https://github.com/user-attachments/assets/0d1fa237-66b7-43ce-8332-fc8244f0097a">

**Reply by Vrodvi** [reply]
В CDA это фиксированные значения, подставляются безусловно.

**Comment by eugenekogan-spb** [comment]
### 2.4 Использованный шаблон templateId
У1-3: Один [1..1] элемент ClinicalDocument/templateId ОБЯЗАН быть представлен со значением атрибута @root равным ”1.2.643.5.1.13.2.7.5.1.7.9.4” (закрытый шаблон).
<img width="766" alt="image" src="https://github.com/user-attachments/assets/4fb18244-3dbc-49a7-b41e-18930fb7c65f">

**Reply by Vrodvi** [reply]
В FHIR не предусмотрено явного указания шаблонов, вместо этого имеется опциональная возможность указать тэг профиля.
См. https://hl7.org/fhir/R5/comparison-cda.html, "Templates and Profiles".

Теоретически можно завести FHIR-профили под каждый исходный шаблон CDA и составить маппинг соответствия для последующей конвертации.

Тут также присутствует более существенная проблема с тем, что в CDA есть понятие "закрытых шаблоны", т.е. такие, в которых автоматически запрещено всё, что явно не разрешено. 
В FHIR закрытые шаблоны явно не приветствуются и их использование не рекомендуется.

**Reply by eugenekogan-spb** [reply]
Если следовать этому предложению, надо завести справочник руководств СЭМД и указывать их в тэге.

```
* meta
  * tag
    * system = "core-cs-cda-guide-id"
    * code = #1.2.643.5.1.13.13.14.7.9.5

CodeSystem: Core_Cs_CDA_Giude_Id
Id: core-cs-cda-guide-id
* #1.2.643.5.1.13.13.14.7.9.5 "Руководство СЭМД ПЛИ версия 5"
```

**Comment by eugenekogan-spb** [comment]
### 2.5 Идентификатор документа id
У1-4: Один [1..1] элемент ClinicalDocument/id ОБЯЗАН быть представлен (атрибут @root должен быть заполнен синтаксически корректным OID, т.е. должен быть сформирован по правилу формирования идентификаторов документов из Таблица 2, т.е. "OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС.51", атрибут @extension должен иметь не пустое значение).
<img width="763" alt="image" src="https://github.com/user-attachments/assets/6c044127-3a07-4516-8384-2ece6e710ca0">

**Reply by eugenekogan-spb** [reply]
В данном случае имеется в виду идентификатор, зависимый от версии, то есть, после замены версии значение идентификатора изменится.
В МИС должен также существовать идентификатор документа, независимый от версии. Он приводится [ниже в атрибуте setId](https://github.com/fhir-ru/core/discussions/268#discussioncomment-10432086)

```
// * identifier[version-dependent-id]
* identifier[+]
  * system = "1.2.643.5.1.13.13.12.2.22.1576.100.1.1.51"
  * value = "7854321"
  * type
    * text = "Внутренний номер документа в МИС, зависимый от версии"
```

**Comment by eugenekogan-spb** [comment]
### 2.6 Код типа документа code
У1-5: Один [1..1] элемент ClinicalDocument/code ОБЯЗАН быть представлен (атрибут @code должен иметь значение "7", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1522", атрибут @codeSystemName должен иметь значение "Виды медицинской документации", атрибуты @codeSystemVersion, @displayName должны быть представлены не пустыми значениями).
<img width="688" alt="image" src="https://github.com/user-attachments/assets/f69154a1-713a-4a9f-9a91-cbe1f87fc728">

**Reply by eugenekogan-spb** [reply]
Аналог в Composition
```
* type = http://loinc.org#11502-2
```

**Reply by eugenekogan-spb** [reply]
Поскольку СЭМД требует указать код по справочнику видов медицинской документации, необходимо расширение.

```
//extension[composition.nsi-document-type]
* extension[+]
  * url                  = "nsi-document-type"
  * valueCodeableConcept = urn:oid:1.2.643.5.1.13.13.11.1522#7    "Протокол лабораторного исследования"
```

**Reply by eugenekogan-spb** [reply]
!!! предложение требует согласования.

**Reply by eugenekogan-spb** [reply]
Решили (встреча 24.12.24) не делать расширение, поскольку CodeableConcept допускает множественное кодирование.
```
* type
  * text = "Протокол лабораторного исследования"
  * coding[+]
    * display = " "
    * system = "[http://loinc.org](http://loinc.org/#11502-2)"
    * code = #11502-2 
  * coding[+]
    * display = "Протокол лабораторного исследования"
    * system = "urn:oid:1.2.643.5.1.13.13.11.1522"
    * version = "8.4"
    * code = #7
```

**Comment by eugenekogan-spb** [comment]
### 2.7 Человекочитаемый заголовок title
У1-6: Один [1..1] элемент ClinicalDocument/title ОБЯЗАН быть представлен и содержать не пустое наполнение.
<img width="439" alt="image" src="https://github.com/user-attachments/assets/e129998d-28f8-4ede-a37f-ccad6fa49552">

**Reply by eugenekogan-spb** [reply]
В Composition

```
* title = "Заголовок документа"
```

**Comment by eugenekogan-spb** [comment]
### 2.8 Дата создания документа effectiveTime
У1-7: Один [1..1] элемент ClinicalDocument/effectiveTime ОБЯЗАН быть представлен (атрибут @value должен быть заполнен согласно общему ограничению "Дата и время").
<img width="515" alt="image" src="https://github.com/user-attachments/assets/1c5d29ea-1f5c-400c-a244-ebe754847913">

**Reply by eugenekogan-spb** [reply]
В Composition

```
* date = "2013-02-01T12:30:02Z"
```
Надо заметить, что название атрибута в СЭМД - дата создания документа, а в Composition - дата последнего редактирования документа. Но поскольку это единственные атрибуты с датой, и они и там, и там обязательны к заполнению, можно считать сопоставление приемлемым и не создавать расширений.

**Reply by Vrodvi** [reply]
Можно ориентироваться на базовый маппинг атрибутов FHIR на CDA, который является частью спецификации FHIR - https://hl7.org/fhir/composition-mappings.html:
![image](https://github.com/user-attachments/assets/15a54a3f-206b-4b97-b752-b5919c62b6f2)

**Comment by eugenekogan-spb** [comment]
### 2.9 Уровень конфиденциальности confidentialityCode
У1-8: Один [1..1] элемент ClinicalDocument/confidentialityCode ОБЯЗАН быть представлен (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.285", атрибут @codeSystemName должен иметь значение "Уровень конфиденциальности медицинского документа", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены не пустыми значениями).
<img width="729" alt="image" src="https://github.com/user-attachments/assets/635cb1c6-5ee1-48e6-a93d-76353413f4a4">

**Reply by eugenekogan-spb** [reply]
Предлагаю сделать расширение. В стандартных расширениях http://hl7.org/fhir/R4/extensibility-registry.html не нашел.
```
//* extension[composition-confidentiality]
  * extension[+]
    * url = "composition-confidentiality"
    * valueCoding
      * code = #N
      * system = "1.2.643.5.1.13.13.99.2.285"
      * version = "1.2"         
      * display = "Обычный"
```

**Comment by eugenekogan-spb** [comment]
### 2.10 Язык документа languageCode
У1-9: Один [1..1] элемент ClinicalDocument/languageCode ОБЯЗАН быть представлен (атрибут @code должен иметь значение "ru-RU").
<img width="277" alt="image" src="https://github.com/user-attachments/assets/3883ffc2-1eaf-4bec-ad75-d4d96e1c8e75">

**Reply by eugenekogan-spb** [reply]
В Composition
```
* language = http://tools.ietf.org/html/bcp47#ru
```

**Comment by eugenekogan-spb** [comment]
### 2.11 Версионность документа setId и versionNumber
У1-10: Элемент ClinicalDocument ОБЯЗАН иметь один [1..1] элемент setId (атрибуты @root, @extension должны быть представлены с не пустыми значениями; атрибут @root должен быть сформирован по правилу формирования идентификаторов наборов версий документов, т.е. "OID_медицинской_организации.100.НомерМИС.НомерЭкзМИС.50", атрибут @extension должен соответствовать идентификатору набора версий документа), содержащий уникальный идентификатор набора версий документа.

У1-11: Элемент ClinicalDocument ОБЯЗАН иметь один [1..1] элемент versionNumber (атрибут @value должен быть представлен с не пустым значением), содержащий номер версии документа.

<img width="532" alt="image" src="https://github.com/user-attachments/assets/90d0fb6c-e30d-47ce-bcde-302db80bfe88" />

Пример указания setId и versionNumber.

**Reply by eugenekogan-spb** [reply]
Здесь предлагается указать идентификатор документа, независимый от версии - как и положено для Composition
```
// * identifier[version-dependent-id]
* identifier[+]
  * system = "1.2.643.5.1.13.13.12.2.22.1576.100.1.1.50"
  * value = "9633"
  * type
    * text = "Внутренний номер документа в МИС, независимый от версии (идентификатор набора версий)"

* version = "1"
```

**Comment by Vrodvi** [comment]
> ### 2.2. Область применения realmCode
> У1-1: Один [1..1] элемент ClinicalDocument/realmCode ОБЯЗАН быть представлен со значением атрибута @code равным "RU". <img alt="image" width="491" src="https://private-user-images.githubusercontent.com/62525329/360978193-b88e8325-eb36-455d-b740-232bd42468a4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU5NjIyMjgsIm5iZiI6MTcyNTk2MTkyOCwicGF0aCI6Ii82MjUyNTMyOS8zNjA5NzgxOTMtYjg4ZTgzMjUtZWIzNi00NTVkLWI3NDAtMjMyYmQ0MjQ2OGE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEwVDA5NTIwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0ZjBhYzc4MzVmZTE4YTBjYTNjMDViNjE4MWI4NTgzMGIwOWQyZGRkNzAxZjk3NGRmNDNjOTFlZGVlNTk0MzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7nZFsJfEwrD3RGj5SWGCcW6EWcLlpKY-izllcPXuow8">



Прямого аналога нет, возможная замена - обозначение через указание российского профиля для ресурса composition.
См. пример из зарубежного руководства https://hl7.org/fhir/us/ccda/StructureDefinition-US-Realm-Header.html

**Reply by eugenekogan-spb** [reply]
Все что мы строим это и есть российский профиль.
Предлагаю пока считать константой.
